(mostly) wi fixes and polish

one-some opened the pull request:
- use icons instead of symbols for buttons
- usage of `Enter` to do things like save and move on from other tags is no longer frustrating
- dont crash when user uploads an image without saving
- dont crash when trying to use robot button on read only
- fix weird tag wrapping behavior
- make welcome text links have underlines (so you can tell they're links without hovering) (these links arent clickable until #415 is fixed due to welcome logo thingey)
- clarify tooltip language, make experimental ui toggle tooltip threatening
- make the robot button noun prompt less strange (this confused a lot of people, whoops)
- reduce code duplication for dealing with wi tags
- other assorted small polish thingeys

known issues:
- naming 2 wi cards the same thing and editing the name causes it to freak out
- somehow im pretty sure we're still duplicating wi cards (unless my test save had them already)
- naming many tags the same thing and deleting one will delete them all. we shouldn't allow multiple of the same tag

unknown issues:
- lots
- repro is hard!!

one-some committed:
WI: Tag polish

- make the button thingeys look more button-ey
- enter now saves the focused tag and focuses the placeholder tag
- reduced code duplication for primary vs secondary and normal vs
  placeholder tags

one-some committed:
WI: Don't explode when user uploads image without a save

This is a band-aid on an underlying problem: there is no save directory
to put blobs like images in before the user saves the game. The way
forward is probably to have an in-memory (no disk, Colab privacy (thats
kind of an oxymoron)) folder or something. I want to expand the data
storage functionality into an api in the future so devs can seamlessly
do something like:

Data.get_file_contents("image/blahblah.png")

and they won't have to actually worry about stuff like this

one-some committed:
Backends: Fix ReadOnly

Since somewhere in the pipeline ReadOnly is ignored, the bug wasn't
actually apparent unless using things like the Robot Button in WI cards.

one-some committed:
WI: Fix visual oddness with more than one row of tags

one-some committed:
UI: Make welcome text links have underlines

Would love to make em' a different color but that would require going
through all the themes and im laaaazy

one-some committed:
UI: Possibly more clear tooltips

one-some committed:
UI: Fix visual inconsistancies in sidebar

one-some committed:
UI: Make experimental tooltip more ominous

The current tooltip made it sound more like "beta features" than "pre
pre pre pre alpha features"

one-some committed:
WI: Enter to save name

one-some committed:
WI: Make the noun thingey more intuitive

one-some committed:
WI: Make delete button pretty

henk717 commented:
Still getting duplicate WI entry errors inside folders, and it can break UI1.
Apparently United does again as well, but its definately not stable yet.

one-some committed:
Merge branch 'united' of https://github.com/henk717/KoboldAI into wi-fixes

one-some committed:
Merge branch 'united' of https://github.com/henk717/KoboldAI into wi-fixes

one-some committed:
WI: Fix UI1 WI errors

one-some commented:
UI1 errors should be fixed

one-some committed:
WI: Workaround for Chrome order weirdness

Chrome fires `blur()` before deleting nodes, meaning the -1 WI was
getting sent after being deleted, resulting in two
`delete_new_world_info_entry` packets being sent to the browser.

Really, it would be better to not do this full WI reset/sync cycle and
just send state changes and update accordingly. That would stop all the
WI weirdness probably.

one-some committed:
UI: Hold shift to skip confirmation dialog

idea stolen from discord, who likely stole it from somebody else

one-some committed:
Merge branch 'united' of https://github.com/henk717/KoboldAI into wi-fixes
